rust-toolchain.toml

```
@@ -3,7 +3,7 @@
 # https://github.com/rust-lang/rustup/issues/2686
 # The auto-installation behaviour in rustup will likely be removed:
 # https://github.com/rust-lang/rustup/issues/1397
-channel = "1.70.0"
+channel = "nightly-2023-06-14"
```

We should keep this as stable. The version here ensures that the stable version is installed and used by default, while we install & use nightly separately for tools that need it.
```suggestion
channel = "1.70.0"
```